addons/cni-canal/canal.yaml

```diff
@@ -4467,16 +4478,27 @@ spec:
             # chosen from this range. Changing this value after installation will have
             # no effect. This should fall within `--cluster-cidr`.
             - name: CALICO_IPV4POOL_CIDR
-              value: "{{ .Config.ClusterNetwork.PodSubnet }}"
+              value: "{{ takeCIDRn 0 .Config.ClusterNetwork.PodSubnet }}"
             # Disable file logging so `kubectl logs` works.
             - name: CALICO_DISABLE_FILE_LOGGING
               value: "true"
             # Set Felix endpoint to host default action to ACCEPT.
             - name: FELIX_DEFAULTENDPOINTTOHOSTACTION
               value: "ACCEPT"
+            {{ if eq .Config.ClusterNetwork.IPFamily "IPv4+IPv6" }}
+            - name: FELIX_IPV6SUPPORT
+              value: "true"
+            - name: IP6
+              value: "autodetect"
+            - name: IP6_AUTODETECTION_METHOD
+              value: "skip-interface=envoyagent"
```

Comment:
Unnecessary in kubeone - there is no envoyagent interface (that is KKP specific)